src/init/features/turtles/repo.ts

```diff
@@ -0,0 +1,184 @@
+): Promise<gcb.Repository> {
+  const connectionId = generateConnectionId(stackId);
+  await getOrCreateConnection(projectId, location, connectionId);
+
+  let remoteUri = await promptRepositoryURI(projectId, location, connectionId);
+  while (remoteUri === "") {
+    await utils.openInBrowser("https://github.com/apps/google-cloud-build/installations/new");
+    await promptOnce({
+      type: "input",
+      message:
+        "Press any key once you have finished configuring your installation's access settings.",
+    });
+    remoteUri = await promptRepositoryURI(projectId, location, connectionId);
+  }
+
+  const repo = await getOrCreateRepository(projectId, location, connectionId, remoteUri);
+  logger.info(`Successfully linked GitHub repository at remote URI ${remoteUri}.`);
+  return repo;
+}
+
+async function promptRepositoryURI(
+  projectId: string,
+  location: string,
+  connectionId: string
+): Promise<string> {
+  const resp = await gcb.fetchLinkableRepositories(projectId, location, connectionId);
+  if (!resp.repositories || resp.repositories.length === 0) {
+    throw new FirebaseError(
+      "The GitHub App does not have access to any repositories. Please configure " +
+        "your app installation permissions at https://github.com/settings/installations."
+    );
+  }
+  const choices = resp.repositories.map((repo: gcb.Repository) => ({
+    name: extractRepoSlugFromURI(repo.remoteUri) || repo.remoteUri,
+    value: repo.remoteUri,
+  }));
+  choices.push({
+    name: "Missing a repo? Select this option to configure your installation's access settings",
+    value: "",
+  });
+
+  return await promptOnce({
+    type: "list",
+    message: "Which of the following repositories would you like to link?",
+    choices,
+  });
+}
+
+async function promptConnectionAuth(
+  conn: gcb.Connection,
+  projectId: string,
+  location: string,
+  connectionId: string
+): Promise<gcb.Connection> {
+  logger.info(conn.installationState.message);
+  logger.info(conn.installationState.actionUri);
+  await utils.openInBrowser(conn.installationState.actionUri);
+  await promptOnce({
+    type: "input",
+    message:
+      "Press any key once you have authorized the app (Cloud Build) to access your GitHub repo.",
+  });
+  return await gcb.getConnection(projectId, location, connectionId);
+}
+
+/**
+ * Exported for unit testing.
+ */
+export async function getOrCreateConnection(
+  projectId: string,
+  location: string,
+  connectionId: string
+): Promise<gcb.Connection> {
+  let conn: gcb.Connection;
+  try {
+    conn = await gcb.getConnection(projectId, location, connectionId);
+  } catch (err: unknown) {
+    if ((err as FirebaseError).status === 404) {
+      const op = await gcb.createConnection(projectId, location, connectionId);
+      conn = await poller.pollOperation<gcb.Connection>({
+        ...gcbPollerOptions,
+        pollerName: `create-${location}-${connectionId}`,
+        operationResourceName: op.name,
+      });
```

Comment:
T poller waits for Promise and resolves when the operation is completed. But can these be used after every await or should these be used in certain usecases(here this is not used with getConnection but used with createConnection).

Reply:
The `getConnection` API simply returns a Connection, so there's no need to poll since there is no Operation to poll on. `createConnection` on the other hand does require polling until the Operation completes.